web-common/src/features/dashboards/pivot/pivot-queries.ts

```diff
@@ -0,0 +1,162 @@
+import type {
+  PivotAxesData,
+  PivotDataStoreConfig,
+} from "@rilldata/web-common/features/dashboards/pivot/types";
+import type { StateManagers } from "@rilldata/web-common/features/dashboards/state-managers/state-managers";
+import { sanitiseExpression } from "@rilldata/web-common/features/dashboards/stores/filter-utils";
+import { useTimeControlStore } from "@rilldata/web-common/features/dashboards/time-controls/time-control-store";
+import type { TimeRangeString } from "@rilldata/web-common/lib/time/types";
+import {
+  V1Expression,
+  V1MetricsViewAggregationDimension,
+  V1MetricsViewAggregationResponseDataItem,
+  V1MetricsViewAggregationSort,
+  createQueryServiceMetricsViewAggregation,
+  type V1MetricsViewAggregationResponse,
+} from "@rilldata/web-common/runtime-client";
+import type { CreateQueryResult } from "@tanstack/svelte-query";
+import { Readable, derived, readable } from "svelte/store";
+
+/**
+ * Wrapper function for Aggregate Query API
+ */
+export function createPivotAggregationRowQuery(
+  ctx: StateManagers,
+  measures: string[],
+  dimensions: V1MetricsViewAggregationDimension[],
+  whereFilter: V1Expression,
+  sort: V1MetricsViewAggregationSort[] = [],
+  limit = "100",
+  offset = "0",
+  timeRange: TimeRangeString | undefined = undefined,
+): CreateQueryResult<V1MetricsViewAggregationResponse> {
+  if (!sort.length) {
+    sort = [
+      {
+        desc: false,
+        name: measures[0] || dimensions?.[0]?.name,
+      },
+    ];
+  }
+
+  return derived(
+    [ctx.runtime, ctx.metricsViewName, useTimeControlStore(ctx)],
+    ([runtime, metricViewName, timeControls], set) =>
+      createQueryServiceMetricsViewAggregation(
+        runtime.instanceId,
+        metricViewName,
+        {
+          measures: measures.map((measure) => ({ name: measure })),
+          dimensions,
+          where: sanitiseExpression(whereFilter, undefined),
+          // TODO: having filter
+          timeStart: timeRange?.start
+            ? timeRange.start
+            : timeControls.timeStart,
+          timeEnd: timeRange?.end ? timeRange.end : timeControls.timeEnd,
```

Comment:
Noting `timeStart` and `timeEnd` are deprecated now in favor of `timeRange`

![image](https://github.com/rilldata/rill/assets/14206386/7a43145e-6bc4-4299-a33c-2ecca4145d7c)